src/JuMP.jl

```diff
@@ -335,21 +335,126 @@ Base.broadcastable(model::Model) = Ref(model)
     backend(model::Model)
 
 Return the lower-level MathOptInterface model that sits underneath JuMP. This
-model depends on which operating mode JuMP is in (see [`mode`](@ref)), and
-whether there are any bridges in the model.
+model depends on which operating mode JuMP is in (see [`mode`](@ref)).
 
-If JuMP is in `DIRECT` mode (i.e., the model was created using
-[`direct_model`](@ref)), the backend will be the optimizer passed to
-[`direct_model`](@ref).
+ * If JuMP is in `DIRECT` mode (i.e., the model was created using
+   [`direct_model`](@ref)), the backend will be the optimizer passed to
+   [`direct_model`](@ref).
+ * If JuMP is in `MANUAL` or `AUTOMATIC` mode, the backend is a
+   `MOI.Utilities.CachingOptimizer`.
 
-If JuMP is in `MANUAL` or `AUTOMATIC` mode, the backend is a
-`MOI.Utilities.CachingOptimizer`.
+**This function should only be used by advanced users looking to access
+low-level MathOptInterface or solver-specific functionality.**
 
-This function should only be used by advanced users looking to access low-level
-MathOptInterface or solver-specific functionality.
+## Notes
+
+If JuMP is not in `DIRECT` mode, the type returned by `backend` may change
+between any JuMP releases. Therefore, only use the public API exposed by
+MathOptInterface, and do not access internal fields such as
+`backend(model).optimizer.model`. If you require access to the
```

Comment:
This example of accessing internal fields isn't necessary, it only suggests users to try it out :)